packages/amplify-category-auth/src/__tests__/provider-utils/awscloudformation/auth-stack-builder/__snapshots__/auth-stack-transform.test.ts.snap

```diff
@@ -786,84 +786,76 @@ exports[`Check Auth Template Generated authstack template in manual flow 1`] = `
-          if (res.DomainDescription && res.DomainDescription.UserPool) {
-            return false;
-          }
-          return true;
-        })
-        .catch((err) => {
-          return false;
-        });
-    };
-    let createUserPoolDomain = (domainName) => {
-      let params = { Domain: domainName, UserPoolId: userPoolId };
-      return identity.createUserPoolDomain(params).promise();
-    };
-    identity
-      .describeUserPool({ UserPoolId: userPoolId })
-      .promise()
-      .then((result) => {
-        if (inputDomainName) {
-          if (result.UserPool.Domain === inputDomainName) {
-            return;
-          } else {
-            if (!result.UserPool.Domain) {
-              return checkDomainAvailability(inputDomainName).then((isDomainAvailable) => {
-                if (isDomainAvailable) {
-                  return createUserPoolDomain(inputDomainName);
-                } else {
-                  throw new Error('Domain not available');
-                }
-              });
-            } else {
-              return checkDomainAvailability(inputDomainName).then((isDomainAvailable) => {
-                if (isDomainAvailable) {
-                  return deleteUserPoolDomain(result.UserPool.Domain).then(() => createUserPoolDomain(inputDomainName));
-                } else {
-                  throw new Error('Domain not available');
-                }
-              });
-            }
-          }
-        } else {
-          if (result.UserPool.Domain) {
-            return deleteUserPoolDomain(result.UserPool.Domain);
-          }
-        }
-      })
-      .then(() => {
-        response.send(event, context, response.SUCCESS, {});
-      })
-      .catch((err) => {
-        console.log(err);
-        response.send(event, context, response.FAILED, { err });
-      });
+  if (inputDomainName) {
+    // create new or replace existing domain.
+    const isDomainAvailable = await checkDomainAvailability(inputDomainName);
+    if (isDomainAvailable) {
+      if (result.UserPool.Domain) {
+        await deleteUserPoolDomain(result.UserPool.Domain, userPoolId);
+      }
+      await createUserPoolDomain(inputDomainName, userPoolId);
+    } else {
+      throw new Error('Domain not available');
+    }
+  } else if (result.UserPool.Domain) {
+    // if input domain is undefined delete existing domain if exists.
+    await deleteUserPoolDomain(result.UserPool.Domain, userPoolId);
   }
-};
+}
+
+async function handleEvent(event, context) {
+  try {
+    const userPoolId = event.ResourceProperties.userPoolId;
+    const inputDomainName = event.ResourceProperties.hostedUIDomainName;
+    if (event.RequestType === 'Delete') {
+      await deleteUserPoolDomain(inputDomainName, userPoolId);
+    } else if (event.RequestType === 'Update' || event.RequestType === 'Create') {
+      await createOrUpdateDomain(inputDomainName, userPoolId);
+    }
+    response.send(event, context, response.SUCCESS, {});
+  } catch (err) {
```

Comment:
nit: it might be more efficient to use `switch` here, but it's ok since there are not many items.